library/psa_crypto.c

```diff
@@ -5111,10 +4898,7 @@ static psa_status_t psa_key_derivation_input_internal(
                                  PSA_ALG_HKDF_GET_HASH( kdf_alg ),
                                  step, data, data_length );
     }
-    else
-#endif /* MBEDTLS_MD_C */
-#if defined(MBEDTLS_MD_C)
-    if( PSA_ALG_IS_TLS12_PRF( kdf_alg ) )
+    else if( PSA_ALG_IS_TLS12_PRF( kdf_alg ) )
```

Comment:
Why did we need to make this change?

Reply:
Actually it's not needed. I just stumbled upon the `endif` followed by an `if` for the same condition and decided to remove them. Afterwards, it turned out we have an orphaned `else` followed by an `if` statement, so I merged them.

If I have missed a reason for the code to be the way it was, I'll gladly revert the change.

Reply:
The unit tests are failing in some builds we return different error codes  than before. I am not certain that this causes the failures, but if it does, then we need to revert it.